beacon_chain/keystore_management.nim

```
@@ -533,14 +632,12 @@ proc importKeystoresFromDir*(rng: var BrHmacDrbgContext,
         continue
 
       let keystore =
-        try:
-          Json.loadFile(file, Keystore)
-        except SerializationError as e:
-          warn "Invalid keystore", err = e.formatMsg(file)
-          continue
-        except IOError as e:
-          warn "Failed to read keystore file", file, err = e.msg
-          continue
+        block:
+          let res = Keystore.loadStore(file)
+          if res.isOk():
+            res.get()
+          else:
+            continue
```

Does invalid keystores lead to silent errors now?

---

`loadKeystore` is private procedure, when it returns `Option.None` consumers of this procedure `validatorKeysFromDirs()` and `validatorKeys` perform `quit 1`. So its not silent.

---

I don't understand. What I meant is that if you have a directory with (let's say) 5 valid and 5 invalid keystores, the code will just ignore the invalid ones without printing any warning or error message (on error, it just performs `continue` and goes to the next keystore).